extensions/common/context.cc

```diff
@@ -295,13 +277,75 @@ flatbuffers::DetachedBuffer extractLocalNodeFlatBuffer() {
   return fbb.Release();
 }
 
+PeerNodeInfo::PeerNodeInfo(const std::string_view peer_metadata_id_key,
+                           const std::string_view peer_metadata_key) {
+  found_ = getValue({peer_metadata_id_key}, &peer_id_);
+  if (found_) {
+    getValue({peer_metadata_key}, &peer_node_);
+    return;
+  }
+  if (peer_metadata_id_key == kDownstreamMetadataIdKey) {
+    // Downstream peer's metadata will never be in localhost endpoint. Skip
+    // looking for it.
+    return;
+  }
+
+  // Construct a fallback peer node metadata based on endpoint labels if it is
+  // not in filter state.
+  std::string endpoint_labels;
+  fallback_peer_node_ = extractEmptyNodeFlatBuffer();
+  if (!getValue(
+          {"upstream_host_metadata", "filter_metadata", "istio", "workload"},
+          &endpoint_labels)) {
+    return;
+  }
+  std::vector<std::string_view> parts = absl::StrSplit(endpoint_labels, ';');
+  // workload label should semicolon separated four parts string:
+  // workload_name;namespace;canonical_service;canonical_revision.
+  if (parts.size() < 4) {
+    return;
+  }
+
+  flatbuffers::FlatBufferBuilder fbb;
+  flatbuffers::Offset<flatbuffers::String> workload_name, namespace_;
```

Comment:
nit: no need of underscore in namespace?

Reply:
namespace is reserved key word, so the generated code suffixed it to work around.